src/cache.ts

```diff
@@ -308,15 +308,22 @@ export interface CachedR2BucketOptions {
 }
 
 export class CachedR2Bucket {
-	private bucket: R2Bucket;
+	private readonly bucket: R2Bucket;
+	private readonly prefix: string;
 	constructor(
-		private ctx: Context,
+		ctx: Context,
 		bucket: R2Bucket,
 		private cache: ReadableCache,
-		private prefix: string,
+		prefix?: string,
 		private ttl_in_ms = DEFAULT_R2_BUCKET_CACHE_TTL_IN_MS
 	) {
-		this.prefix = prefix;
+		if (prefix?.length === 0) {
```

Comment:
nits: setting prefix before doing check would make them more readable by avoiding `?.`

Reply:
I prefer setting after checking. I know it's the same thing but to me it makes sense that I only assign to `this` when I've guaranteed it's a valid value